src/libraries/System.Net.Sockets/src/System/Net/Sockets/UnixDomainSocketEndPoint.cs

```diff
@@ -44,6 +44,11 @@ private UnixDomainSocketEndPoint(string path, string? boundFileName)
                 bufferLength++;
             }
 
+            if (!Socket.OSSupportsUnixDomainSockets)
+            {
+                throw new PlatformNotSupportedException();
+            }
```

Comment:
why did you move this?

Reply:
we should also annotate the type itself as UnsupportedOSPlatform(iOS/tvOS) and SupportedOSPlatform(MacCatalyst) in the ref

doesn't need to be done in this PR since it'll likely cause quite a bit of fallout

Reply:
I moved it so it would throw PNSE early instead of ArgumentOutOfRangeException on platforms which return a small value for `s_nativePathLength`, like iOSSimulator and tvOSSimulator

Reply:
Never mind, I guess - Windows tests were explicitly looking for the fail-on-path-before-failing-on-PNSE model. I think it's a bad model, but it's the one already in place.